sktree/ensemble/_honest_forest.py

```diff
@@ -31,41 +31,67 @@ def _parallel_build_trees(
     n_samples_bootstrap=None,
     missing_values_in_feature_mask=None,
     classes=None,
+    stratify=False,
 ):
     """
     Private function used to fit a single tree in parallel.
 
-    XXX: this is copied over from scikit-learn and modified to allow sampling with
+    XXX:
+    1. this is copied over from scikit-learn and modified to allow sampling with
     and without replacement given ``bootstrap``.
+
+    2. Overrides the scikit-learn implementation to allow for stratification during bootstrapping
+    via the `stratify` parameter.
```

Comment:
Just confirm, now with parameter 'stratify' = True, will data be stratified twice, including in Bootstrap, and when split into structure set and honest set?

Reply:
Yes, stratification should occur every time we sample the dataset whether its subsampling, or bootstrapping.

1) when we bootstrap sample the dataset to get the in-bag and oob samples, we stratify using `sklearn.utils.resample` (this PR)
2) when we split the in-bag unique samples in halves to get structure and honest dataset, we stratify using StratifiedShuffleSplit.